PhpAmqpLib/Wire/IO/StreamIO.php

```diff
@@ -194,14 +263,42 @@ public function write($data)
                 throw new AMQPTimeoutException('Error sending data. Socket connection timed out');
             }
 
-            $len = $len - $written;
-            if ($len > 0) {
-                $data = mb_substr($data, 0 - $len, 0 - $len, 'ASCII');
-            } else {
-                $this->last_write = microtime(true);
-                break;
-            }
+            $written += $buffer;
+            $data = mb_substr($data, $buffer, mb_strlen($data, 'ASCII') - $buffer, 'ASCII');
         }
+
+        $this->last_write = microtime(true);
+        return;
+    }
+
+    /**
+     * Internal error handler to deal with stream and socket errors that need to be ignored
+     * 
+     * @param  int $errno
+     * @param  string $errstr
+     * @param  string $errfile
+     * @param  int $errline
+     * @param  array $errcontext
+     * @return void
+     */
+    public function error_handler($errno, $errstr, $errfile, $errline, $errcontext = null)
+    {
+        $this->last_error = compact('errno', 'errstr', 'errfile', 'errline', 'errcontext');
+
+        // fwrite notice that the stream isn't ready
+        if (strstr($errstr, 'Resource temporarily unavailable')) {
+             // it's allowed to retry
+             return;
+        }
+
+        // stream_select warning that it has been interrupted by a signal
+        if (strstr($errstr, 'Interrupted system call')) {
+             // it's allowed while processing signals
+             return;
+        }
+
+        // raise all other issues to exceptions
+        throw new \ErrorException($errstr, 0, $errno, $errfile, $errline);
```

Comment:
You are right, sorry about this. I've added a note here: https://github.com/videlalvaro/php-amqplib/blob/master/CHANGELOG.md